src/prover_slow.h

```diff
@@ -100,6 +101,19 @@ std::vector<uint8_t> ProveSlow(integer& D, form& x, uint64_t num_iterations) {
         }
         nudupl_form(y, y, D, L);
         reducer.reduce(y);
+
+        // Check for cancellation every 65535 interations
+        if ((i&0xffff)==0) {
+            // Only if we have a shutdown path
+            if (shutdown_file_path!="") {
+                struct stat buffer;
+            
+                if (stat (shutdown_file_path.c_str(), &buffer) != 0) {
```

Comment:
there's a slight risk that this call fails with `EINTR`, which probably should not be interpreted as a request to abort.

Reply:
fixing this thanks!